Source/Core/InputCommon/ControllerInterface/CemuHookUDPServer/CemuHookUDPServer.cpp

```diff
@@ -0,0 +1,355 @@
+
+  AddInput(new AnalogInput("L2", m_pad_data.trigger_l2, 255));
+  AddInput(new AnalogInput("R2", m_pad_data.trigger_r2, 255));
+
+  AddInput(new Button("L3", m_pad_data.button_states1, 0x2));
+  AddInput(new Button("R3", m_pad_data.button_states1, 0x4));
+  AddInput(new Button("Share", m_pad_data.button_states1, 0x1));
+  AddInput(new Button("Options", m_pad_data.button_states1, 0x8));
+  AddInput(new Button("PS", m_pad_data.button_ps, 0x1));
+  AddInput(new Button("Touch Button", m_pad_data.button_touch, 0x1));
+
+  AddInput(new AnalogInput("Left X-", m_pad_data.left_stick_x, -128, -128));
+  AddInput(new AnalogInput("Left X+", m_pad_data.left_stick_x, 127, -128));
+  AddInput(new AnalogInput("Left Y-", m_pad_data.left_stick_y_inverted, -128, -128));
+  AddInput(new AnalogInput("Left Y+", m_pad_data.left_stick_y_inverted, 127, -128));
+  AddInput(new AnalogInput("Right X-", m_pad_data.right_stick_x, -128, -128));
+  AddInput(new AnalogInput("Right X+", m_pad_data.right_stick_x, 127, -128));
+  AddInput(new AnalogInput("Right Y-", m_pad_data.right_stick_y_inverted, -128, -128));
+  AddInput(new AnalogInput("Right Y+", m_pad_data.right_stick_y_inverted, 127, -128));
+
+  AddInput(new AnalogInput("Touch X-", m_touch_x, -TOUCH_X_AXIS_MAX));
+  AddInput(new AnalogInput("Touch X+", m_touch_x, TOUCH_X_AXIS_MAX));
+  AddInput(new AnalogInput("Touch Y-", m_touch_y, -TOUCH_Y_AXIS_MAX));
+  AddInput(new AnalogInput("Touch Y+", m_touch_y, TOUCH_Y_AXIS_MAX));
+
+  AddInput(new AnalogInput("Accelerometer Left", m_accel.x, -WiimoteEmu::ACCELEROMETER_MAX));
+  AddInput(new AnalogInput("Accelerometer Right", m_accel.x, WiimoteEmu::ACCELEROMETER_MAX));
+  AddInput(new AnalogInput("Accelerometer Backward", m_accel.y, -WiimoteEmu::ACCELEROMETER_MAX));
+  AddInput(new AnalogInput("Accelerometer Forward", m_accel.y, WiimoteEmu::ACCELEROMETER_MAX));
+  AddInput(new AnalogInput("Accelerometer Up", m_accel.z, -WiimoteEmu::ACCELEROMETER_MAX));
+  AddInput(new AnalogInput("Accelerometer Down", m_accel.z, WiimoteEmu::ACCELEROMETER_MAX));
+
+  AddInput(new AnalogInput("Gyro Pitch Up", m_gyro.x, -WiimoteEmu::GYROSCOPE_MAX));
+  AddInput(new AnalogInput("Gyro Pitch Down", m_gyro.x, WiimoteEmu::GYROSCOPE_MAX));
+  AddInput(new AnalogInput("Gyro Roll Right", m_gyro.y, -WiimoteEmu::GYROSCOPE_MAX));
+  AddInput(new AnalogInput("Gyro Roll Left", m_gyro.y, WiimoteEmu::GYROSCOPE_MAX));
+  AddInput(new AnalogInput("Gyro Yaw Right", m_gyro.z, -WiimoteEmu::GYROSCOPE_MAX));
+  AddInput(new AnalogInput("Gyro Yaw Left", m_gyro.z, WiimoteEmu::GYROSCOPE_MAX));
+}
+
+std::string Device::GetName() const
+{
+  switch (m_model)
+  {
+  case Proto::DsModel::None:
+    return "None";
+  case Proto::DsModel::DS3:
+    return "DualShock 3";
+  case Proto::DsModel::DS4:
+    return "DualShock 4";
+  case Proto::DsModel::Generic:
+    return "Generic Gamepad";
+  default:
+    return "Device";
+  }
+}
+
+std::string Device::GetSource() const
+{
+  return "CemuHookUDPServer";
```

Comment:
This is maybe a little too verbose for the device list?

Reply:
Maybe just "UDPServer" would be better, considering the reasons that have been brought up to not name it after Cemuhook?

Reply:
I agree. I'll change it to "UDPServer" as suggested, hoping there will not be other UDP controller input protocols in the future.